test/e2e/storage/volume_provisioning.go

```diff
@@ -408,6 +414,39 @@ var _ = SIGDescribe("Dynamic Provisioning", func() {
 			}
 		})
 
+		It("should provision storage with different reclaim policy", func() {
+			framework.SkipUnlessProviderIs("gce", "gke")
+
+			test := storageClassTest{
+				"HDD PD on GCE/GKE",
+				[]string{"gce", "gke"},
+				"kubernetes.io/gce-pd",
+				map[string]string{
+					"type": "pd-standard",
+				},
+				"1Gi",
+				"1Gi",
+				func(volume *v1.PersistentVolume) error {
+					return checkGCEPD(volume, "pd-standard")
+				},
+			}
+			class := newStorageClass(test, ns, "reclaimPolicy")
+			retain := v1.PersistentVolumeReclaimRetain
+			class.ReclaimPolicy = &retain
+			claim := newClaim(test, ns, "reclaimPolicy")
+			claim.Spec.StorageClassName = &class.Name
+			pv := testDynamicProvisioning(test, c, claim, class)
+
+			By(fmt.Sprintf("waiting for the provisioned PV %q to enter phase %s", pv.Name, v1.VolumeReleased))
+			framework.ExpectNoError(framework.WaitForPersistentVolumePhase(v1.VolumeReleased, c, pv.Name, 1*time.Second, 30*time.Second))
+
+			By(fmt.Sprintf("deleting the storage asset backing the PV %q", pv.Name))
+			framework.ExpectNoError(framework.DeletePDWithRetry(pv.Spec.GCEPersistentDisk.PDName))
+
+			By(fmt.Sprintf("deleting the PV %q", pv.Name))
+			framework.ExpectNoError(framework.DeletePersistentVolume(c, pv.Name), "Failed to delete PV ", pv.Name)
```

Comment:
It is clearer to delete the pv here but curious as to why you delete it here when it could have been unconditionally deleted in `testDynamicProvisioning()`

Reply:
imo it makes sense because the manual deletion is cloudprovider-specific, and outside of the scope of the kube dynamic provisioning process, so i kept it out of testDynamicProvisioning

Reply:
I agree.

Reply:
okay, wouldn't hurt: will add WaitForPersistentVolumeDeleted after this line